# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/policy_target.cc
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 2
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/policy_target.cc
--- a/security/sandbox/chromium/sandbox/win/src/policy_target.cc	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/policy_target.cc	Wed Jun 12 11:10:48 2019 +0100
@@ -25,8 +25,8 @@
 SANDBOX_INTERCEPT NtExports g_nt;
 
 // Policy data.
-extern void* volatile     g_shared_policy_memory;
-SANDBOX_INTERCEPT size_t  g_shared_policy_size;
+extern void* volatile g_shared_policy_memory;
+SANDBOX_INTERCEPT size_t g_shared_policy_size;
 
 bool QueryBroker(int ipc_id, CountedParameterSetBase* params) {
   DCHECK_NT(static_cast<size_t>(ipc_id) < kMaxServiceCount);
@@ -53,7 +53,7 @@
     return false;
   }
 
-  for (int i = 0; i < params->count; i++) {
+  for (size_t i = 0; i < params->count; i++) {
     if (!params->parameters[i].IsValid()) {
       NOTREACHED_NT();
       return false;
@@ -61,8 +61,8 @@
   }
 
   PolicyProcessor processor(policy);
-  PolicyResult result = processor.Evaluate(kShortEval, params->parameters,
-                                           params->count);
+  PolicyResult result =
+      processor.Evaluate(kShortEval, params->parameters, params->count);
   DCHECK_NT(POLICY_ERROR != result);
 
   return POLICY_MATCH == result && ASK_BROKER == processor.GetAction();
@@ -73,8 +73,10 @@
 // Hooks NtSetInformationThread to block RevertToSelf from being
 // called before the actual call to LowerToken.
 NTSTATUS WINAPI TargetNtSetInformationThread(
-    NtSetInformationThreadFunction orig_SetInformationThread, HANDLE thread,
-    NT_THREAD_INFORMATION_CLASS thread_info_class, PVOID thread_information,
+    NtSetInformationThreadFunction orig_SetInformationThread,
+    HANDLE thread,
+    NT_THREAD_INFORMATION_CLASS thread_info_class,
+    PVOID thread_information,
     ULONG thread_information_bytes) {
   do {
     if (SandboxFactory::GetTargetServices()->GetState()->RevertedToSelf())
@@ -95,32 +97,37 @@
     return STATUS_SUCCESS;
   } while (false);
 
-  return orig_SetInformationThread(thread, thread_info_class,
-                                   thread_information,
-                                   thread_information_bytes);
+  return orig_SetInformationThread(
+      thread, thread_info_class, thread_information, thread_information_bytes);
 }
 
 // Hooks NtOpenThreadToken to force the open_as_self parameter to be set to
-// FALSE if we are still running with the impersonation token. open_as_self set
-// to TRUE means that the token will be open using the process token instead of
+// false if we are still running with the impersonation token. open_as_self set
+// to true means that the token will be open using the process token instead of
 // the impersonation token. This is bad because the process token does not have
 // access to open the thread token.
-NTSTATUS WINAPI TargetNtOpenThreadToken(
-    NtOpenThreadTokenFunction orig_OpenThreadToken, HANDLE thread,
-    ACCESS_MASK desired_access, BOOLEAN open_as_self, PHANDLE token) {
+NTSTATUS WINAPI
+TargetNtOpenThreadToken(NtOpenThreadTokenFunction orig_OpenThreadToken,
+                        HANDLE thread,
+                        ACCESS_MASK desired_access,
+                        BOOLEAN open_as_self,
+                        PHANDLE token) {
   if (!SandboxFactory::GetTargetServices()->GetState()->RevertedToSelf())
-    open_as_self = FALSE;
+    open_as_self = false;
 
   return orig_OpenThreadToken(thread, desired_access, open_as_self, token);
 }
 
 // See comment for TargetNtOpenThreadToken
-NTSTATUS WINAPI TargetNtOpenThreadTokenEx(
-    NtOpenThreadTokenExFunction orig_OpenThreadTokenEx, HANDLE thread,
-    ACCESS_MASK desired_access, BOOLEAN open_as_self, ULONG handle_attributes,
-    PHANDLE token) {
+NTSTATUS WINAPI
+TargetNtOpenThreadTokenEx(NtOpenThreadTokenExFunction orig_OpenThreadTokenEx,
+                          HANDLE thread,
+                          ACCESS_MASK desired_access,
+                          BOOLEAN open_as_self,
+                          ULONG handle_attributes,
+                          PHANDLE token) {
   if (!SandboxFactory::GetTargetServices()->GetState()->RevertedToSelf())
-    open_as_self = FALSE;
+    open_as_self = false;
 
   return orig_OpenThreadTokenEx(thread, desired_access, open_as_self,
                                 handle_attributes, token);